cpp/src/io/utilities/hostdevice_vector.hpp

```diff
@@ -164,13 +164,15 @@ class hostdevice_vector {
   /**
    * @brief Converts a part of a hostdevice_vector into a hostdevice_span.
    *
-   * @return A typed hostdevice_span of the hostdevice_vector's data.
+   * @param offset The offset of the first element in the subspan
+   * @param count The number of elements in the subspan
+   * @return A typed hostdevice_span of the hostdevice_vector's data
    */
   [[nodiscard]] hostdevice_span<T> subspan(size_t offset, size_t count)
   {
     CUDF_EXPECTS(count >= offset, "End index cannot be smaller than the starting index.");
     CUDF_EXPECTS(count <= d_data.size(), "Slice range out of bounds.");
```

Comment:
The naming of `count` is a bit misleading. According to the above code, it seems `offset` indicates the starting index and `count` refers to the end index. So `count - offset` should be correct. @ttnghia does your change break any existing unit tests?

Reply:
No there were not any relevant unit tests for that function. I just added tests for it.

Oh these checks are also invalid. I'm going to update them.